rust/src/lib/ifaces/inter_ifaces.rs

```diff
@@ -176,6 +179,47 @@ impl Interfaces {
         Ok(())
     }
 
+    fn remove_unknown_type_port(&mut self) {
+        let mut pending_actions: Vec<(String, InterfaceType, String)> =
+            Vec::new();
+        for (iface_name, iface) in self.kernel_ifaces.iter() {
+            if !iface.is_controller() {
+                continue;
+            }
+            for port_name in find_unknown_type_port(iface, self) {
+                pending_actions.push((
+                    iface_name.to_string(),
+                    iface.iface_type(),
+                    port_name.to_string(),
+                ));
+            }
+        }
+        for iface in self.user_ifaces.values() {
+            if !iface.is_controller() {
+                continue;
+            }
+            for port_name in find_unknown_type_port(iface, self) {
+                pending_actions.push((
+                    iface.name().to_string(),
+                    iface.iface_type(),
+                    port_name.to_string(),
+                ));
+            }
+        }
+
+        for (ctrl_name, ctrl_type, port_name) in pending_actions {
+            if ctrl_type.is_userspace() {
+                if let Some(iface) =
+                    self.user_ifaces.get_mut(&(ctrl_name, ctrl_type))
+                {
+                    iface.remove_port(&port_name)
+                }
+            } else if let Some(iface) = self.kernel_ifaces.get_mut(&ctrl_name) {
+                iface.remove_port(&port_name)
+            }
+        }
```

Comment:
Looking here it might not be necessary to have those two for loops and then this loop down here. You could remove the ports directly without intermediate `Vec`.

Reply:
Not sure what you are referring. The reason I use `pending_actions` is because we should not change item during iterating even that is safe to human being.